src/Sulu/Bundle/WebsiteBundle/Routing/ContentRouteProvider.php

```diff
@@ -232,12 +232,15 @@ private function checkResourceLocator($resourceLocator, $resourceLocatorPrefix)
      */
     protected function getRedirectRoute(Request $request, $url)
     {
+        $requestFormat = $request->getRequestFormat();
+        $format = 'html' !== $requestFormat ? '.' . $requestFormat : '';
+
         // redirect to linked page
         return new Route(
             $this->decodePathInfo($request->getPathInfo()),
             [
                 '_controller' => 'SuluWebsiteBundle:Redirect:redirect',
-                'url' => $url,
+                'url' => $url . $format,
```

Comment:
have not tested this, but maybe we could also set `_format` similar to `_controller`? https://symfony.com/doc/current/routing.html#special-parameters

maybe we would not need to handle the `html` case then because symfony would do that for us. but not sure about this

Reply:
Ah, I could try this of course. I'll let you know.

Reply:
I've tried adding `_format` and removing the format from the `url`, but unfortunately it does not work as expected. It does not add `.json` to the redirected URL. I would suggest leaving it like it is, unless you guys can come up with a better solution.